ClemBot.Bot/bot/utils/helpers.py

```diff
@@ -24,28 +25,58 @@ def format_datetime(time: datetime) -> str:
     return time.strftime("%Y-%m-%dT%H:%M:%S.%f")
 
 
-def format_duration(duration: FutureDuration | PastDuration) -> str:
+def format_duration(
+    duration: Annotated[datetime | relativedelta, FutureDuration | PastDuration]
+) -> str:
     """
     Formats the given datetime to a string.
     Uses relativedelta to calculate the difference between datetime.utcnow()
     and the given datetime and formats it as:
     A Year(s) B Month(s) C Week(s) D Day(s) E Hour(s) F Minute(s) G Second(s)
     """
-    now = datetime.utcnow()
-    delta = relativedelta(now, duration) if duration < now else relativedelta(duration, now)  # type: ignore
-    s = ""
-    if delta.years > 0:
-        s += f'{delta.years} Year{"s" if delta.years > 1 else ""} '
-    if delta.months > 0:
-        s += f'{delta.months} Month{"s" if delta.months > 1 else ""} '
-    if delta.weeks > 0:
-        s += f'{delta.weeks} Week{"s" if delta.weeks > 1 else ""} '
-    if delta.days > 0:
-        s += f'{delta.days} Day{"s" if delta.days > 1 else ""} '
-    if delta.hours > 0:
-        s += f'{delta.hours} Hour{"s" if delta.hours > 1 else ""} '
-    if delta.minutes > 0:
-        s += f'{delta.minutes} Minute{"s" if delta.minutes > 1 else ""} '
-    if delta.seconds > 0:
-        s += f'{delta.seconds} Second{"s" if delta.seconds > 1 else ""}'
-    return s
+    if isinstance(duration, datetime):
+        if duration < datetime.utcnow():
+            delta = relativedelta(datetime.utcnow(), duration)
+        else:
+            delta = relativedelta(duration, datetime.utcnow())
+    else:  # duration isinstance relativedelta
+        delta = duration
+        duration = datetime.utcnow() + duration
+    arrow_date = arrow.get(duration)
+    granularity = _get_timedelta_granularity(delta, 3)
+    return arrow_date.humanize(only_distance=True, granularity=granularity)  # type: ignore
```

Comment:
Why is there a `type: ignore` here?

Reply:
mypy - expects `Literal['year', 'month', 'week', 'day', 'hour', 'minute', 'second']` and we give `list[str]`

Reply:
sus